gui-whelktool/src/main/resources/templates/change-complex-subject.groovy

```diff
@@ -0,0 +1,77 @@
+PrintWriter failedBibIDs = getReportWriter("failed-to-update-bibIDs")
+PrintWriter scheduledForUpdate = getReportWriter("scheduled-for-update")
+PrintWriter manualReviewLog = getReportWriter("needs-manual-review")
+
+selectBySqlWhere( "collection = 'bib' AND (" +
+        "data#>'{@graph,1,instanceOf,subject}' @> '[{\"@type\":\"ComplexSubject\"}]'::jsonb OR " +
+        "data#>'{@graph,1,subject}' @> '[{\"@type\":\"ComplexSubject\"}]'::jsonb" +
+        ")",
+        silent: false, { bib ->
+    boolean changed = false
+
+    if (bib.graph[1].instanceOf != null &&
+            bib.graph[1].instanceOf.subject != null &&
+            bib.graph[1].instanceOf.subject instanceof List) {
+        for (Object subject : bib.graph[1].instanceOf.subject) {
+            changed |= fixSubject(subject, manualReviewLog, bib.doc.shortId)
+        }
+    }
+
+    if (bib.graph[1].subject != null && bib.graph[1].subject instanceof List) {
+        for (Object subject : bib.graph[1].subject) {
+            changed |= fixSubject(subject, manualReviewLog, bib.doc.shortId)
+        }
+    }
+
+    if (changed) {
+        scheduledForUpdate.println("${bib.doc.getURI()}")
+        bib.scheduleSave(loud: true, onError: { e ->
+            failedBibIDs.println("Failed to update ${bib.doc.shortId} due to: $e")
+        })
+    }
+})
+
+boolean fixSubject(Object subject, PrintWriter manualReviewLog, String id) {
+    boolean fixedSameAs = false
+    boolean fixedPrefLabel = false
+    boolean fixedTermComponentList = false
+
+    if (subject["@type"] == "ComplexSubject") {
+
+        if (subject.sameAs != null && subject.sameAs instanceof List) {
```

Comment:
Could we have any sameAs that is a Map?

Reply:
> Could we have any sameAs that is a Map?

Om inget ändrats radikalt, så ska det vara omöjligt (har varit omöjligt tidigare i alla fall).